libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/models/StepIntersection.kt

```diff
@@ -0,0 +1,130 @@
+package org.maplibre.navigation.android.navigation.v5.models
+
+import com.google.gson.annotations.SerializedName
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.maplibre.geojson.Point
+import org.maplibre.navigation.android.navigation.v5.models.serializer.PointSerializer
+
+/**
+ * Object representing an intersection along the step.
+ *
+ * @since 1.3.0
+ */
+@Serializable
+data class StepIntersection(
+
+    /**
+     * A [Point] representing this intersection location.
+     *
+     * @since 3.0.0
+     */
+    @Serializable(with = PointSerializer::class)
+    val location: Point,
+
+    /**
+     * An integer list of bearing values available at the step intersection.
+     *
+     * @since 1.3.0
+     */
+    val bearings: List<Int>?,
+
+    /**
+     * A list of strings signifying the classes of the road exiting the intersection. Possible
+     * values:
+     *
+     *  * **toll**: the road continues on a toll road
+     *  * **ferry**: the road continues on a ferry
+     *  * **restricted**: the road continues on with access restrictions
+     *  * **motorway**: the road continues on a motorway
+     *  * **tunnel**: the road continues on a tunnel
+     *
+     *
+     * @since 3.0.0
+     */
+    val classes: List<String>?,
+
+    /**
+     * A list of entry flags, corresponding in a 1:1 relationship to the bearings. A value of true
+     * indicates that the respective road could be entered on a valid route. false indicates that the
+     * turn onto the respective road would violate a restriction.
+     *
+     * @since 1.3.0
+     */
+    val entry: List<Boolean>?,
+
+    /**
+     * Index into bearings/entry array. Used to calculate the bearing before the turn. Namely, the
+     * clockwise angle from true north to the direction of travel before the maneuver/passing the
+     * intersection. To get the bearing in the direction of driving, the bearing has to be rotated by
+     * a value of 180. The value is not supplied for departure
+     * maneuvers.
+     *
+     * @since 1.3.0
+     */
+    @SerialName("in")
+    val inIndex: Int?,
+
+    /**
+     * Index out of the bearings/entry array. Used to extract the bearing after the turn. Namely, The
+     * clockwise angle from true north to the direction of travel after the maneuver/passing the
+     * intersection. The value is not supplied for arrive maneuvers.
+     *
+     * @since 1.3.0
+     */
+    @SerialName("out")
+    val outIndex: Int?,
+
+    /**
+     * Array of lane objects that represent the available turn lanes at the intersection. If no lane
+     * information is available for an intersection, the lanes property will not be present. Lanes are
+     * provided in their order on the street, from left to right.
+     *
+     * @since 2.0.0
+     */
+    val lanes: List<IntersectionLanes?>?,
```

Comment:
Does it make sense to return a list of `null`s in this context?

Reply:
Fixed ✅